src/Serialization/Rings.jl

```diff
@@ -0,0 +1,111 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
+    
+    return t(modulus)
+end
+
+function save_internal(s::SerializerState, R::Nemo.NmodRing)
+    return Dict(
+        :modulus => save_type_dispatch(s, modulus(R))
+    )
+end
+
+################################################################################
+#  Polynomial Rings
+function save_internal(s::SerializerState, R::Union{MPolyRing, PolyRing})
+    return Dict(
+        :symbols => save_type_dispatch(s, symbols(R)),
+        :base_ring => save_type_dispatch(s, base_ring(R)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       T::Type{<: Union{MPolyRing, PolyRing}},
+                       dict::Dict)
+    base_ring = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+    symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols]) 
+
+    if T isa PolyRing
+        return PolynomialRing(base_ring, symbols...)
+    end
+
+    return PolynomialRing(base_ring, symbols)
+end
+
+################################################################################
+# Multivariate Polynomials
+function save_internal(s::SerializerState, p::MPolyElem)
+    parent_ring = parent(p)
+    parent_ring = save_type_dispatch(s, parent_ring)
+    terms = []
+    
+    for (i, c) in enumerate(coeffs(p))
+        term = Dict(
+            :exponent => save_type_dispatch(s, exponent_vector(p, i)),
+            :coeff => save_type_dispatch(s, c)
```

Comment:
Might as well avoid that iterator and do this:
```suggestion
    for i in 1:length(p)
        term = Dict(
            :exponent => save_type_dispatch(s, exponent_vector(p, i)),
            :coeff => save_type_dispatch(s, coeff(p, i))
```